operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/InformerWrapper.java

```diff
@@ -25,6 +26,18 @@
 
   public InformerWrapper(SharedIndexInformer<T> informer) {
     this.informer = informer;
+
+    // register
+    ConfigurationServiceProvider.instance().getInformerStoppedHandler()
+        .ifPresent(ish -> {
+          final var stopped = informer.stopped();
+          if (stopped != null) {
```

Comment:
in which condition `stopped` is going to be `null` ?
With the current implementation in the `client` it doesn't seem to be possible.

I would leave this check out to fail early if this erroneous situation appears.

Reply:
Good point. This happens during tests, though, where it would require quite a bit of setup just to mock the future…

Reply:
I would rather prefer more setup in the tests as opposed to the risk of this to happen and be swallowed in a real operator.
The risk here is to end up with a non-working callback without noticing.

If you are really, really against doing it, I should, at least, ask for a loud warning when this situation occurs .